src/coreclr/jit/gentree.cpp

```diff
@@ -20608,7 +20644,8 @@ GenTree* Compiler::gtNewSimdBinOpNode(
                     }
                     else if (simdSize == 16 && compOpportunisticallyDependsOn(InstructionSet_AVX2))
                     {
-                        if (IsBaselineVector512IsaSupportedOpportunistically())
+                        if (IsBaselineVector512IsaSupportedOpportunistically() ||
```

Comment:
I think we can just use `canUseEvexEncoding()` instead or maybe have a more explicit helper API that clarifies the intent here.

The main intent of `IsBaselineVector512IsaSupportedOpportunistically()` was to abstract away the dependence on `F+BW+CD+DQ+VL` into a single API, while the new check is just `(F+BW+CD+DQ+VL) || Avx10v1`

So maybe something like `IsAvx512OrAvx10SupportedOpportunistically()` would be sufficient.

Reply:
In places like below
```
if (IsBaselineVector512IsaSupportedOpportunistically() ||
                            (isAvx10v1 = compOpportunisticallyDependsOn(InstructionSet_AVX10v1)))
                        {
                            // Input is SIMD16 [U]Byte and AVX512BW_VL is supported:
                            // - Widen inputs as SIMD32 [U]Short
                            // - Multiply widened inputs (SIMD32 [U]Short) as widened product (SIMD32 [U]Short)
                            // - Narrow widened product (SIMD32 [U]Short) as SIMD16 [U]Byte
                            widenIntrinsic = NI_AVX2_ConvertToVector256Int16;

                            if (simdBaseType == TYP_BYTE)
                            {
                                widenedSimdBaseJitType = CORINFO_TYPE_SHORT;
                                narrowIntrinsic        = isAvx10v1 ? NI_AVX10v1_ConvertToVector128SByte : NI_AVX512BW_VL_ConvertToVector128SByte;
                            }
```

we would end up requiring 2 ```compOpportunisticallySupports``` checks in cases where ```Avx10v1``` is supported and ``AVX512F``` is not. is that okay? I was trying to minimize the number of ```compOpportunisticallySupports``` checks mainly by using a bool ```isAvx10v1```. Do we rely on compiler optimization here?

Reply:
The key thing is that `compOpportunisticallyDependsOn` actually tracks the ISA check in scenarios like the R2R output binary. So we need to be careful in the overall usage and how cached lookups are reused.

We could maybe have some `bool compIsEvexOpportunisticallySupported(bool* isV512Supported)` and then it could be used like:
```cpp
bool isV512Supported = false;

if (compIsEvexOpportunisticallySupported(&isV512Supported))
{
    // Input is SIMD16 [U]Byte and AVX512BW_VL is supported:
    // - Widen inputs as SIMD32 [U]Short
    // - Multiply widened inputs (SIMD32 [U]Short) as widened product (SIMD32 [U]Short)
    // - Narrow widened product (SIMD32 [U]Short) as SIMD16 [U]Byte
    widenIntrinsic = NI_AVX2_ConvertToVector256Int16;

    if (simdBaseType == TYP_BYTE)
    {
        widenedSimdBaseJitType = CORINFO_TYPE_SHORT;
        narrowIntrinsic        = !isV512Supported ? NI_AVX10v1_ConvertToVector128SByte : NI_AVX512BW_VL_ConvertToVector128SByte;
    }
}
```

Which works given:
```cpp
bool compIsEvexOpportunisticallySupported(bool& isV512Supported)
{
    if (IsBaselineVector512IsaSupportedOpportunistically())
    {
        isV512Supported = true;
        return true;
    }

    isV512Supported = false;
    return compOpportunisticallyDependsOn(InstructionSet_AVX10v1);
}
```

I think that makes it a bit more explicit/clear in the usage and how the checks go together.

Reply:
I did not think it this way. This makes sense. I have made the suggested changes.